src/main/scala/edu/ie3/simona/service/ev/ExtEvDataService.scala

```diff
@@ -222,70 +220,91 @@ class ExtEvDataService(override val scheduler: ActorRef)
     )
   }
 
-  private def sendEvMovements(
+  private def requestDepartingEvs(
       tick: Long,
-      evMovements: java.util.Map[UUID, EvcsMovements]
+      allDepartingEvs: java.util.Map[UUID, java.util.List[UUID]]
   )(implicit
       serviceStateData: ExtEvStateData
   ): (ExtEvStateData, Option[Seq[ScheduleTriggerMessage]]) = {
-    val filteredPositionData =
-      evMovements.asScala.flatMap { case (evcs, movements) =>
+
+    allDepartingEvs.asScala.foreach { case (evcs, departingEvs) =>
+      serviceStateData.uuidToActorRef.get(evcs) match {
+        case Some(evcsActor) =>
+          evcsActor ! DepartingEvsRequest(tick, departingEvs.asScala.toSeq)
+        case None =>
+          log.warning(
+            "A corresponding actor ref for UUID {} could not be found",
+            evcs
+          )
+      }
+    }
+
+    val departingEvResponses: Map[UUID, Option[Seq[EvModel]]] =
+      serviceStateData.uuidToActorRef.map { case (evcs, _) =>
+        evcs -> None
+      }
+
+    // if there are no departing evs during this tick,
+    // we're sending response right away
```

Comment:
Question: `departingEvResponses.isEmpty` in this simulation would only be true if the `uuidToActorRef` map would be empty. Wouldn't this mean there is no evcs in the simulation instead of there are no departing evs during this tick? 

Did you mean to check `if(allDepartingEvs.isEmpty)` ?

Reply:
You're right, this is not right